### PR TITLE
[movie] Extracting all the frames of a movie into a tile

### PR DESCRIPTION
**Pull request by Fabian467**
**Problem**
There was no way of generating a tile from a movie

**Solution**
Added function to create and store the tile

### CODE REVIEW

File: zou/app/blueprints/previews/__init__.py [hunk 1]
```diff
@@ -112,6 +113,10 @@
         "/actions/preview-files/<preview_file_id>/update-annotations",
         UpdateAnnotationsResource,
     ),
+    (
+        "/data/preview-files/<preview_file_id>/extract-frame",
```
Comment (EvanBldy) [inline zou/app/blueprints/previews/__init__.py:117]: It should be extract-tile, no ?

File: zou/utils/movie.py [hunk 2]
```diff
@@ -118,10 +118,32 @@ def get_all_frames(movie_path):
...
+def generate_tile(movie_path, movie_fps):
     """
-    ffmpeg -i {movie_path} -vf 'scale=150:100,tile=8x8' -an -vsync 0 tile%03d.png
+    Generates a tile from a movie.
     """
+    folder_path = os.path.dirname(movie_path)
+    file_source_name = os.path.basename(movie_path)
+    file_target_name = f"{file_source_name[:-4]}_tile.png"
+    file_target_path = os.path.join(folder_path, file_target_name)
+    command = [
```
Comment (EvanBldy) [inline zou/utils/movie.py:129]: Please use the ffmpeg wrapper (https://pypi.org/project/ffmpeg-python) to do that. 
https://stackoverflow.com/questions/31024968/using-ffmpeg-to-obtain-video-durations-in-python

File: zou/utils/movie.py [hunk 3]
```diff
@@ -118,10 +118,32 @@ def get_all_frames(movie_path):
...
+    ]
+    #w,h = get_movie_size(movie_path)
+
+    output = subprocess.check_output(command, universal_newlines=True)
+    duration_in_seconds = float(output)
+
+    float_movie_fps = float(movie_fps)
+    duration_in_frames = int(duration_in_seconds * float_movie_fps)
+    rows = math.ceil(duration_in_frames / 8)
+    command = f'ffmpeg -i "{movie_path}" -vf "fps=1,scale=720:480,tile=8x{rows}" {file_target_name}'
```
Comment (EvanBldy) [inline zou/utils/movie.py:144]: Please use the ffmpeg wrapper (https://pypi.org/project/ffmpeg-python) to do that.

**Comment by frankrousseau** [comment]
I close it in favor of https://github.com/cgwire/zou/pull/700

Thank you @Fabian467 for your contribution.

**Pull request closed without merging**